tutorial/westeros/westeros.ipynb

```diff
@@ -0,0 +1,1268 @@
+    "\n",
+    "# in GWh - from IEA Electricity Output\n",
+    "activity = {\n",
+    "    'coal_ppl': 0.5 * total_demand,\n",
+    "    'wind_ppl': 0.5 * total_demand,\n",
+    "}\n",
+    "\n",
+    "for tec, val in activity.items():\n",
+    "    df = make_df(base_activity, technology=tec, value=val)\n",
+    "    scenario.add_par('bound_activity_up', df)\n",
+    "    scenario.add_par('bound_activity_lo', df)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Objective Function"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "source": [
+    "The objective function seeks to minimize total discounted system cost over space and time. \n",
+    "\n",
+    "$\\min \\sum_{n,y \\in Y^{M}} interestrate_{y} \\cdot COST\\_NODAL_{n,y}$\n",
+    "\n",
+    "\n",
+    "Let's add the interest rate parameter."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "rate = [0.05] * len(horizon)\n",
+    "unit = ['%'] * len(horizon)\n",
+    "scenario.add_par(\"interestrate\", key=horizon, val=rate, unit=unit)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "source": [
+    "However, `COST_NODAL` is comprised of a variety of costs related to the use of different technologies."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "### Investment Costs\n",
+    "\n",
+    "$inv\\_cost_{n,t,y} \\cdot construction\\_time\\_factor_{n,t,y} \\cdot end\\_of\\_horizon\\_factor_{n,t,y} \\cdot CAP\\_NEW_{n,t,y}$"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 20,
+   "metadata": {
+    "slideshow": {
+     "slide_type": "skip"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# TODO hide this\n",
+    "base_inv_cost = {\n",
+    "    'node_loc': country,\n",
+    "    'year_vtg': horizon,\n",
+    "    'unit': 'USD/GWa',\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# in $ / kW\n",
+    "costs = {\n",
+    "    'coal_ppl': 1500,\n",
+    "    'wind_ppl': 1100,\n",
+    "    'bulb': 5,\n",
+    "}\n",
+    "\n",
+    "for tec, val in costs.items():\n",
+    "    df = make_df(base_inv_cost, technology=tec, value=val * 1e6)\n",
+    "    scenario.add_par('inv_cost', df)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "### Fixed O&M Costs\n",
```

Comment:
You could add a line like 'Fixed cost are only relevant as long as the capacity is active. This formulation allows to include the potential cost savings from early retirement of installed capacity.'

Reply:
done!